File: c++/human.cpp
/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//memanggil header yang dibutuhkan
#include <iostream>
#include <string.h>

using namespace std;

//membuat class human dan memanggil class parentnya yaitu mahasiswa
class human : public mahasiswa{
    //membuat atribut private
	private:
        string nik;

    public:
        //membuat konstruktor tanpa parameter
        human(){
        }

        //membuat konstruktordengan parameter
        human(string nik){
            this->nik = nik;
        }

        //membuat fungsi untuk mengeset atribut private
        void set_nik(string nik){
            this->nik = nik;
        }

        //membuat fungsi untuk me-return atribut private
        string get_nik(){
            return nik;
        }

        //membuat destruktor
        ~human(){
        }
};

File: php/Human.php
<?php

/*
    Saya Mohammad Ray Mosaid mengerjakan Latihan 2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
    Aamiin.
*/

//membuat class Human dengan menunjuk parentnya yaitu Mahasiswa
class Human extends Mahasiswa{
    //membuat atribut private
	private $nik = "";

    //membuat konstruktor
    function __construct(){
    }

    //membuat fungsi untuk mengeset atribut private
    function setNik($nik){
        $this->nik = $nik;
    }

    //membuat fungsi untuk me-return atribut private
    function getNik(){
        return $this->nik;
    }

    //membuat destruktor
    function __destruct(){
    }
}
?>
